activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -703,12 +703,19 @@ def rewhere(conditions)
     # argument.
     #
     # The two relations must be structurally compatible: they must be scoping the same model, and
-    # they must differ only by #where (if no #group has been defined) or #having (if a #group is
-    # present). Neither relation may have a #limit, #offset, or #distinct set.
+    # they must differ only by #where (if no #group has been defined), #having (if a #group is
+    # present), #annotate or #optimizer_hints. Neither relation may have a #limit, #offset, or #distinct set.
```

Comment:
I'm not excited to talk about the `annotate` and the `optimizer_hints` in the same breath with the `where` and the `having`.

We had hit the structually incompatible error on several occasions in the past (#29461, ea6139101ccaf8be03b536b1293a9f36bc12f2f7), in that time, we just relaxed the limitation to address the error, but didn't mention about what values are allowed/ignored.

I agree that there is room for improvement in the doc, but how about discussing it in another PR about how to improve it?

Reply:
Thank you for telling me related pull requests.
I understood.

I reverted the changes in doc and force-pushed.